java/pattern/constraint/ThingConstraint.java

```diff
@@ -387,16 +388,19 @@ public Has(final String type, final UnboundVariable var) {
             this(hidden().type(type), var.toThing());
         }
 
+        public Has(final UnboundVariable var) {this(null, var.toThing());}
+
         private Has(final TypeVariable type, final ThingVariable<?> attribute) {
-            if (type == null || attribute == null) throw new NullPointerException("Null type/attribute");
+            if (attribute == null) throw new NullPointerException("Null attribute");
             this.type = type;
             this.attribute = attribute.constrain(new Isa(type, false));
             this.hash = Objects.hash(Has.class, this.type, this.attribute);
         }
 
-        public ThingVariable<?> attribute() {
-            return attribute;
-        }
+
+        public ThingVariable<?> attribute() { return attribute; }
+
+        public TypeVariable type() { return type; }
```

Comment:
lets add @Nullable for the return type here

Reply:
Done

Reply:
We don't do `@Nullable` for return type @flyingsilverfin , @maxbaxt . We only do You may notice from the rest of the code base that we only do `@Nullable` for method parameters, and if a return type of a getter could possibly be `null`, then we wrap it in a `Optional` -- which is what `Optional` is made for.